vector/src/main/java/im/vector/app/features/location/LocationTracker.kt

```diff
@@ -50,34 +50,28 @@ class LocationTracker @Inject constructor(
             return
         }
 
-        val isGpsEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)
-        val isNetworkEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
-
-        Timber.d("## LocationTracker. isGpsEnabled: $isGpsEnabled - isNetworkEnabled: $isNetworkEnabled")
-
-        val provider = when {
-            isGpsEnabled     -> LocationManager.GPS_PROVIDER
-            isNetworkEnabled -> LocationManager.NETWORK_PROVIDER
-            else             -> {
-                callback?.onLocationProviderIsNotAvailable()
-                Timber.v("## LocationTracker. There is no location provider available")
-                return
-            }
-        }
-
-        // Send last known location without waiting location updates
-        locationManager.getLastKnownLocation(provider)?.let { lastKnownLocation ->
-            Timber.d("## LocationTracker. lastKnownLocation")
-            callback?.onLocationUpdate(lastKnownLocation.toLocationData())
-        }
-
-        Timber.d("## LocationTracker. track location using $provider")
-        locationManager.requestLocationUpdates(
-                provider,
-                MIN_TIME_TO_UPDATE_LOCATION_MILLIS,
-                MIN_DISTANCE_TO_UPDATE_LOCATION_METERS,
-                this
-        )
+        locationManager.allProviders
+                .takeIf { it.isNotEmpty() }
+                ?.forEach { provider ->
+                    Timber.d("## LocationTracker. track location using $provider")
+
+                    // Send last known location without waiting location updates
+                    locationManager.getLastKnownLocation(provider)?.let { lastKnownLocation ->
+                        Timber.d("## LocationTracker. lastKnownLocation")
+                        callback?.onLocationUpdate(lastKnownLocation.toLocationData())
+                    }
+
+                    locationManager.requestLocationUpdates(
```

Comment:
Is it OK to call this multiple times this `fun` and call only once `locationManager?.removeUpdates(this)` in `onStop`? Maybe yes but I want to be sure.

Reply:
Yes, you can see the location icon on Status Bar is disappearing when fragment is destroyed.

Reply:
Ah didn't notice that. It does not disappear on my side

Reply:
(I have to put the entire app into bg to make it disappear to be more precise.